# BE131_JK
Berkeley 2018 Fall  BioE131 | lab, class repo

**Introduction to Computational Molecular and Cell Biology**

**Lab List**  
* lab2 : Biological gene tree analysis / bar plotting, box plotting    [link](https://github.com/dgggit/BE131_JK/blob/master/lab02/lab2_data.ipynb)
* lab3 : Raw file analysis from 'Romeo and Juliet.txt' [link](https://github.com/dgggit/BE131_JK/blob/master/lab03/lab3_report.ipynb)
* lab4 : Forming BIO databases with SQlite, making assosiative table [link](https://github.com/dgggit/BE131_JK/blob/master/lab04/Submission_lab4/database_sql.ipynb)
* lab5 : Use RNAfold to predict the folded structure of allosteric ribozyme, verify the bio-logic-gate controlled by OBS binding input & stem II activation output [link](https://github.com/dgggit/BE131_JK/blob/master/lab05/lab5_submission.ipynb)
* lab6 : Understanding the sequences, plot the depth results [link](https://github.com/dgggit/BE131_JK/blob/master/lab06/lab_6.ipynb)
* lab7 : Generating random DNA sequences, run compression algorithms and analyze/calculate based on principle of each algorithm [link](https://github.com/dgggit/BE131_JK/blob/master/lab07/lab7_submission.ipynb)
* lab8 & 9 : Sequence alignment with SPAdes, finding N50 score, Genomic Annotation [link](https://github.com/dgggit/BE131_JK/blob/master/lab0809/lab8_submission.ipynb)
* lab10 : Implementing Needleman-Wunsch algorithm and run-traceback pairwise alignment of amino acids. [link](https://github.com/dgggit/BE131_JK/blob/master/lab10/nw_aligner.py)
* **Final Project : To analyze unknown genome : An analysis of Population Diversity and Genetic Variation [link](https://github.com/dgggit/BE131_JK/tree/master/final_project)**

## License Information
See the [LICENSE](LICENSE) file for license rights and limitations.
