{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sqlite3\n",
    "conn = sqlite3.connect('lab4.db')\n",
    "\n",
    "c = conn.cursor()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# c.execute('''DROP TABLE gene;''')\n",
    "# c.execute('''DROP TABLE enzyme;''')\n",
    "# c.execute('''DROP TABLE en_pa;''')\n",
    "# c.execute('''DROP TABLE pathway;''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<sqlite3.Cursor at 0x7fa2dcf00ea0>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "c.execute('''CREATE TABLE gene\n",
    "             (id INT, ncbi_id TEXT, name_desc TEXT, organism TEXT, chr TEXT, start INT, end INT, eid INT, sequence TEXT )''')\n",
    "\n",
    "c.execute('''CREATE TABLE enzyme\n",
    "            (id INT, name TEXT, function TEXT, ecnum TEXT )''')\n",
    "\n",
    "c.execute('''CREATE TABLE en_pa (eid INT, pid INT, priority INT )''')\n",
    "\n",
    "c.execute('''CREATE TABLE pathway\n",
    "             (id INT, name TEXT, description TEXT )''')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add pathway\n",
    "\n",
    "pa_file = open(\"lab4_pathway_data\")\n",
    "\n",
    "name = ''\n",
    "description = ''\n",
    "\n",
    "iterator = 1\n",
    "\n",
    "for i, line in enumerate(pa_file):\n",
    "\n",
    "    line = line.replace('\\n', '')\n",
    "    \n",
    "    if '#' in line:\n",
    "        continue\n",
    "        \n",
    "    if i%3 is 0:\n",
    "        name = line\n",
    "    elif i%3 is 1:\n",
    "        description = line\n",
    "    elif i%3 is 2:\n",
    "        c.execute(\"INSERT INTO pathway (id, name, description) VALUES (\"+ str(iterator) + \", '\" + name + \"', '\" + description + \"' )\" )        \n",
    "        iterator+=1\n",
    "\n",
    "conn.commit()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add enzyme\n",
    "\n",
    "enz_file = open(\"lab4_enzyme_data\")\n",
    "\n",
    "name = ''\n",
    "pathorder = []\n",
    "function = ''\n",
    "ECnum = ''\n",
    "\n",
    "iterator = 1\n",
    "\n",
    "\n",
    "for i, line in enumerate(enz_file):\n",
    "\n",
    "    line = line.replace('\\n', '')\n",
    "    \n",
    "    if '#' in line:\n",
    "        continue\n",
    "    if i%5 is 0:\n",
    "        name = line\n",
    "    elif i%5 is 1:\n",
    "        pathorder = line.split(',')\n",
    "        for i, item in enumerate(pathorder):\n",
    "            pathorder[i] = pathorder[i].split('|')\n",
    "            pathorder[i][1] = int(pathorder[i][1])\n",
    "    elif i%5 is 2:\n",
    "        function = line\n",
    "    elif i%5 is 3:\n",
    "        ECnum = line\n",
    "    elif i%5 is 4:\n",
    "        c.execute(\"INSERT INTO enzyme (id, name, function, ecnum) VALUES (\"+ str(iterator) + \", '\" + name + \"', '\" + function +\"', '\"+ ECnum + \"' )\" )\n",
    "        \n",
    "        ### TODO ###\n",
    "        for rep in range(len(pathorder)):\n",
    "            c.execute(\"INSERT INTO en_pa (eid, pid, priority) VALUES (\"+ str(iterator) + \",\" + \"(select id from pathway where name = \"+\"'\"+pathorder[rep][0]+\"')\"+\" ,\"+str(pathorder[rep][1])+\")\")\n",
    "            \n",
    "        iterator+=1\n",
    "        \n",
    "conn.commit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #Search for Genes: Drosophila_melanogaster : Drosophila_melanogaster_sequen.fasta\n",
    "# from Bio import Entrez\n",
    "# from Bio import SeqIO\n",
    "# from Bio.SeqRecord import SeqRecord\n",
    "\n",
    "# #Enzyme List:\n",
    "# #               Triosephosphate isomerase   \n",
    "# #                 Phosphoglycerate kinase   \n",
    "# #         Phosphoenolpyruvate carboxylase   \n",
    "# #                         Pyruvate kinase   \n",
    "# #       Glucose-6-phosphate dehydrogenase   \n",
    "# #                        Gluconolactonase   \n",
    "# #        6-phosphogluconate dehydrogenase   \n",
    "# #                           Transaldolase   \n",
    "# #                Succinate dehydrogenase   \n",
    "# #                               Fumarase   \n",
    "# #                  Malate dehydrogenase   \n",
    "# #                      Citrate synthase\n",
    "\n",
    "# # Possibble db\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# Entrez.email = 'mike.ferna512@berkeley.edu'\n",
    "# handle = Entrez.esearch(db='nucleotide',\n",
    "#                        term='E coli Triosephosphate isomerase',\n",
    "#                        sort='relevance',\n",
    "#                        idtype='acc')\n",
    "\n",
    "# for i in Entrez.read(handle)['IdList']:\n",
    "#     handle = Entrez.efetch(db='nucleotide',id = i, rettype='fasta', retmode='text', retmax = 1)\n",
    "#     ids = i;\n",
    "#     break\n",
    "# # make sure to view sequence before adding to database to be sure it is the correct sequence\n",
    "# handle = Entrez.efetch(db=\"nucleotide\", id= ids, rettype=\"fasta\", retmode=\"text\")\n",
    "# record = handle.read()\n",
    "# print(record)\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# #----------------------#\n",
    "# # Add sequence\n",
    "    \n",
    "# #record1 = open('Drosophila_melanogaster_sequen.fasta', 'a+')\n",
    "# record1 = open('E_Coli_sequen.fasta', 'a+')\n",
    "# #record1 = open('Homo_sapiens_sequen.fasta', 'a+')\n",
    "\n",
    "# handle = Entrez.efetch(db=\"nucleotide\", id= ids, rettype=\"fasta\", retmode=\"text\")\n",
    "# record = handle.read()\n",
    "# print(record)\n",
    "# record1.write(record) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add Gene\n",
    "\n",
    "from Bio import SeqIO\n",
    "import pandas as pd\n",
    "\n",
    "\n",
    "gene_file = ['Drosophila_melanogaster_sequen.fasta', 'E_Coli_sequen.fasta', 'Homo_sapiens_sequen.fasta']\n",
    "org_list = ['Drosophila melanogaster', 'Escherichia coli', 'Homo sapiens']\n",
    "enz_list =    ['Triose phosphate isomerase' ,\n",
    "                'Phosphoglycerate kinase'  ,\n",
    "        'Phosphoenolpyruvate carboxylase'  ,\n",
    "                        'Pyruvate kinase'  ,\n",
    "      'Glucose-6-phosphate dehydrogenase'  ,\n",
    "                       'Gluconolactonase'  ,\n",
    "       '6-phosphogluconate dehydrogenase'  ,\n",
    "                          'Transaldolase' ,\n",
    "               'Succinate dehydrogenase' ,\n",
    "                              'Fumarase', \n",
    "                 'Malate dehydrogenase',\n",
    "                     'Citrate synthase' ]\n",
    "\n",
    "chrRule = ['X', 'Y', 'R', 'L', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0']\n",
    "# c.execute('''CREATE TABLE gene\n",
    "#              (id INT, ncbi_id TEXT, name_desc TEXT, organism TEXT, chr TEXT, start INT, end INT, eid INT, sequence TEXT )''')\n",
    "\n",
    "iterator = 1\n",
    "for file in gene_file:\n",
    "    for gene in SeqIO.parse(file, 'fasta'):\n",
    "        \n",
    "    # analysing information from FASTA\n",
    "        \n",
    "    # 1. id : iterating gene\n",
    "        order_id = iterator\n",
    "        \n",
    "    # 2. ncbi_id : ncbi gene id\n",
    "        ncbi_id = gene.id\n",
    "        \n",
    "    # 3. name_desc : name, description from fasta\n",
    "        name_desc = gene.description[len(gene.id)+1: : ]\n",
    "        \n",
    "    # 4. organism\n",
    "        organism = ''\n",
    "        for org in org_list:\n",
    "            if org in gene.description:\n",
    "                organism = org\n",
    "                break\n",
    "                \n",
    "    # 5. chromosome : (blank if not stated)\n",
    "        chromosome = ''\n",
    "        try:\n",
    "            scan = str(gene.description)\n",
    "            idx = scan.index('chromosome')\n",
    "        \n",
    "            while( scan[idx+11] in chrRule ):\n",
    "                chromosome+=scan[idx+11]\n",
    "                idx+=1\n",
    "        except:\n",
    "            pass\n",
    "        \n",
    "    # 6,7. st, en : starting / position in the gene\n",
    "        st, en = '', ''\n",
    "        \n",
    "        try:\n",
    "            scan = str(gene.name)\n",
    "            idx = scan.index(':c')\n",
    "            \n",
    "            while( scan[idx+2] in chrRule ):\n",
    "                st+=scan[idx+2]\n",
    "                idx+=1\n",
    "            idx+=1\n",
    "            while( scan[idx+2] in chrRule ):\n",
    "                en+=scan[idx+2]\n",
    "                idx+=1\n",
    "                if idx+2 is len(scan):\n",
    "                    break\n",
    "            st = int(st)\n",
    "            en = int(en)\n",
    "            \n",
    "        except:\n",
    "            st=0\n",
    "            en=0\n",
    "            pass\n",
    "\n",
    "\n",
    "        \n",
    "    # 8. eid : match enzyme id from the db\n",
    "        eid = 0\n",
    "        \n",
    "        scan = str(gene.description)\n",
    "        \n",
    "        #print(scan)\n",
    "        for enz in enz_list:\n",
    "            if scan.lower().replace(' ', '').find(enz.lower().replace(' ','')) is not -1:\n",
    "                input_str = enz[0].upper() + enz[1:].lower()\n",
    "                output = str(pd.read_sql_query(\"SELECT id FROM enzyme WHERE name==\"+\"'\"+str(input_str)+\"'\"+ \";\", conn ))\n",
    "\n",
    "                tmp = []\n",
    "                idx = -1\n",
    "                while( output[idx] is not ' '):\n",
    "                    tmp.append(output[idx])\n",
    "                    idx-=1\n",
    "\n",
    "                tmp.reverse()\n",
    "                eid = int(\"\".join(tmp))\n",
    "                break\n",
    "                \n",
    "    #9. seq : nt-sequence\n",
    "        \n",
    "        seq = str(gene.seq)\n",
    "        \n",
    "    # now, lets put to the db!\n",
    "    \n",
    "        c.execute(\"\"\"INSERT INTO gene\n",
    "             (id , ncbi_id , name_desc , organism , chr , start, end, eid, sequence )\n",
    "             VALUES (\"\"\"\n",
    "                 + str(order_id) + ', '\n",
    "                 + \"'\" + ncbi_id + \"', \"\n",
    "                 + \"'\" + name_desc + \"', \"\n",
    "                 + \"'\" + organism + \"', \"\n",
    "                 + \"'\" + chromosome + \"', \"\n",
    "                 + str(st) + ', '\n",
    "                 + str(en) + ', '\n",
    "                 + str(eid)+ ', '\n",
    "                 + \"'\" + seq + \"'\"\n",
    "             \n",
    "             +')'\n",
    "        )\n",
    "        \n",
    "\n",
    "        iterator+=1\n",
    "        \n",
    "    conn.commit()\n",
    "\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    id                            ncbi_id  \\\n",
      "0    1     NT_033777.3:c30134599-30132981   \n",
      "1    2       NT_033779.5:c2753068-2746880   \n",
      "2    3         NZ_KB233235.1:c23004-20365   \n",
      "3    4                        NM_079724.3   \n",
      "4    5                         AH002543.2   \n",
      "5    6      NC_004354.4:12012377-12017375   \n",
      "6    7                           M80598.1   \n",
      "7    8                        NM_079111.3   \n",
      "8    9                        NM_057862.5   \n",
      "9   10                     NM_001298874.1   \n",
      "10  11       NC_000913.3:c4111507-4110740   \n",
      "11  12       NC_000913.3:c3072622-3071459   \n",
      "12  13       NC_000913.3:c4153098-4150447   \n",
      "13  14        NC_000913.3:1755698-1757110   \n",
      "14  15       NC_011751.1:c2171506-2170031   \n",
      "15  16       NC_000913.3:c2101268-2099862   \n",
      "16  17        NC_000913.3:2578666-2579616   \n",
      "17  18          NC_000913.3:755907-757673   \n",
      "18  19       NC_000913.3:c1686588-1685185   \n",
      "19  20       NC_000913.3:c3384268-3383330   \n",
      "20  21         NC_000913.3:c754468-753185   \n",
      "21  22     NC_000023.11:78104169-78126827   \n",
      "22  23    NC_000015.10:c72231624-72199029   \n",
      "23  24  NC_000023.11:c154547586-154531390   \n",
      "24  25                        NM_006755.1   \n",
      "25  26             NG_012340.1:5001-40449   \n",
      "26  27     NC_000007.14:76048019-76067508   \n",
      "27  28    NC_000012.12:c56300391-56271699   \n",
      "\n",
      "                                            name_desc  \\\n",
      "0   Drosophila melanogaster chromosome 3R Triose p...   \n",
      "1   Drosophila melanogaster chromosome 2L Phosphog...   \n",
      "2   Phosphoenolpyruvate carboxylase Drosophila mel...   \n",
      "3   Drosophila melanogaster pyruvate kinase, trans...   \n",
      "4   Drosophila melanogaster glucose-6-phosphate de...   \n",
      "5   Drosophila melanogaster chromosome X: Gluconol...   \n",
      "6   Drosophila melanogaster 6-phosphogluconate deh...   \n",
      "7   Drosophila melanogaster transaldolase (Taldo),...   \n",
      "8   Drosophila melanogaster succinate dehydrogenas...   \n",
      "9   Drosophila melanogaster malate dehydrogenase 1...   \n",
      "10  Escherichia coli str. K-12 substr. E coli Trio...   \n",
      "11  pgk phosphoglycerate kinase [ Escherichia coli...   \n",
      "12  ppc phosphoenolpyruvate carboxylase [ Escheric...   \n",
      "13  pykF pyruvate kinase I [ Escherichia coli str....   \n",
      "14  zwf glucose-6-phosphate dehydrogenase [ Escher...   \n",
      "15  gnd 6-phosphogluconate dehydrogenase, decarbox...   \n",
      "16  talA transaldolase A [ Escherichia coli str. K...   \n",
      "17  sdhA succinate dehydrogenase, flavoprotein sub...   \n",
      "18  umC fumarate hydratase (fumarase C),aerobic Cl...   \n",
      "19  malate dehydrogenase, NAD(P)-binding [ Escheri...   \n",
      "20  Citrate synthase [ Escherichia coli str. K-12 ...   \n",
      "21  Homo sapiens chromosome X, GRCh38.p12 Primary ...   \n",
      "22  Homo sapiens chromosome 15, pyruvate kinase M1...   \n",
      "23  Homo sapiens chromosome X,Glucose-6-phosphate ...   \n",
      "24        Homo sapiens transaldolase 1 (TALDO1), mRNA   \n",
      "25  Homo sapiens succinate dehydrogenase complex i...   \n",
      "26  Homo sapiens chromosome 7,Malate dehydrogenase...   \n",
      "27  Homo sapiens chromosome 12, Citrate synthase, ...   \n",
      "\n",
      "                   organism chr      start        end  eid  \\\n",
      "0   Drosophila melanogaster  3R   30134599   30132981    1   \n",
      "1   Drosophila melanogaster  2L    2753068    2746880    2   \n",
      "2   Drosophila melanogaster          23004      20365    3   \n",
      "3   Drosophila melanogaster              0          0    4   \n",
      "4   Drosophila melanogaster              0          0    5   \n",
      "5   Drosophila melanogaster   X          0          0    6   \n",
      "6   Drosophila melanogaster              0          0    7   \n",
      "7   Drosophila melanogaster              0          0    8   \n",
      "8   Drosophila melanogaster              0          0    9   \n",
      "9   Drosophila melanogaster              0          0   11   \n",
      "10         Escherichia coli        4111507    4110740    1   \n",
      "11         Escherichia coli        3072622    3071459    2   \n",
      "12         Escherichia coli        4153098    4150447    3   \n",
      "13         Escherichia coli              0          0    4   \n",
      "14         Escherichia coli        2171506    2170031    5   \n",
      "15         Escherichia coli        2101268    2099862    7   \n",
      "16         Escherichia coli              0          0    8   \n",
      "17         Escherichia coli              0          0    9   \n",
      "18         Escherichia coli        1686588    1685185   10   \n",
      "19         Escherichia coli        3384268    3383330   11   \n",
      "20         Escherichia coli         754468     753185   12   \n",
      "21             Homo sapiens   X          0          0    2   \n",
      "22             Homo sapiens  15   72231624   72199029    4   \n",
      "23             Homo sapiens   X  154547586  154531390    5   \n",
      "24             Homo sapiens              0          0    8   \n",
      "25             Homo sapiens   1          0          0    9   \n",
      "26             Homo sapiens   7          0          0   11   \n",
      "27             Homo sapiens  12   56300391   56271699   12   \n",
      "\n",
      "                                             sequence  \n",
      "0   AAGCCGTTTATATAATTTCAGAATAACTTTTACTACTCCTTACTTA...  \n",
      "1   GCCGCGGCATTTGTACAGCCGCAGTCGATAAGACAACAACAACAGC...  \n",
      "2   ATGAATCAGCAATATTCCGCAATGCGCAGTAATGTCAGTATGTTGG...  \n",
      "3   AGTCGAACTTTTTGTCTCGTCCTCGCCGCGTCCGATTTTCCATATA...  \n",
      "4   CAGCTGCGCAGTCGGGGTTACAAAGCGTCTGCTGCTTTAGCTGTTG...  \n",
      "5   GTGTAGAAGGTGGCAGCCCTTTAAGGCACCAGCTGACCCCATAGAG...  \n",
      "6   CTCGAGCAGTTCAAGTTCCTGAAGTGAGTTGCGCCACCTTTGTCTT...  \n",
      "7   TATTGCATTGAAGTTGGGCAATGGGAAGTGATCGCACGTTAAAAAA...  \n",
      "8   AGCGAGTCAGTGAACCGAACTGCAGTGCCGACGAGCGGCAGTTCAC...  \n",
      "9   AAAAAGTAGTGACTGGAAAATAATTGTGAAAAGTAACAGGTTCCCA...  \n",
      "10  ATGCGACATCCTTTAGTGATGGGTAACTGGAAACTGAACGGCAGCC...  \n",
      "11  ATGTCTGTAATTAAGATGACCGATCTGGATCTTGCTGGGAAACGTG...  \n",
      "12  ATGAACGAACAATATTCCGCATTGCGTAGTAATGTCAGTATGCTCG...  \n",
      "13  ATGAAAAAGACCAAAATTGTTTGCACCATCGGACCGAAAACCGAAT...  \n",
      "14  ATGGCGGTAACGCAAACAGCCCAGGCCTGTGACCTGGTCATTTTCG...  \n",
      "15  ATGTCCAAGCAACAGATCGGCGTAGTCGGTATGGCAGTGATGGGAC...  \n",
      "16  ATGAACGAGTTAGACGGCATCAAACAGTTCACCACTGTCGTGGCAG...  \n",
      "17  ATGAAATTGCCAGTCAGAGAATTTGATGCAGTTGTGATTGGTGCCG...  \n",
      "18  ATGAATACAGTACGCAGCGAAAAAGATTCGATGGGGGCGATTGATG...  \n",
      "19  ATGAAAGTCGCAGTCCTCGGCGCTGCTGGCGGTATTGGCCAGGCGC...  \n",
      "20  ATGGCTGATACAAAAGCAAAACTCACCCTCAACGGGGATACAGCTG...  \n",
      "21  GAGAGCAGCGGCCGGGAAGGGGCGGTGCGGGAGGCGGGGTGTGGGG...  \n",
      "22  AACCCATAAATCTGGGCCCTGCCCAGGTAGGCCGGGACAGCTGGGG...  \n",
      "23  ACAAGCCCCGCCCCCGAGAACTCGGGAAGCCGGCGAGAAGTGTGAG...  \n",
      "24  CGCGCCCGTCCCGTCGCCGCCGCCGCCGCCGCAGACCCCTCGGTCT...  \n",
      "25  ACCCGGGAAACCGGAAGCCGCCTCCCACTTGGTTGCTCGTACGCGG...  \n",
      "26  TCCCGGAGGCGAGCGCGACCGGAAGTCCGGTCACTCTCGCGAGGCC...  \n",
      "27  AGTGGGCGGGGCCTCCTTGAGGACCCCGGGCTGGGCGCCGCCGCCG...  \n",
      "    id                               name  \\\n",
      "0    1         Triose phosphate isomerase   \n",
      "1    2            Phosphoglycerate kinase   \n",
      "2    3    Phosphoenolpyruvate carboxylase   \n",
      "3    4                    Pyruvate kinase   \n",
      "4    5  Glucose-6-phosphate dehydrogenase   \n",
      "5    6                   Gluconolactonase   \n",
      "6    7   6-phosphogluconate dehydrogenase   \n",
      "7    8                      Transaldolase   \n",
      "8    9            Succinate dehydrogenase   \n",
      "9   10                           Fumarase   \n",
      "10  11               Malate dehydrogenase   \n",
      "11  12                   Citrate synthase   \n",
      "\n",
      "                                             function        ecnum  \n",
      "0   catalyzes the reversible interconversion of th...   EC 5.3.1.1  \n",
      "1   catalyzes the reversible transfer of a phospha...   EC 2.7.2.3  \n",
      "2   an enzyme in the family of carboxy-lyases foun...  EC 4.1.1.31  \n",
      "3   It catalyzes the transfer of a phosphate group...  EC 2.7.1.40  \n",
      "4   D-glucose 6-phosphate + NADP⁺ ⇌ 6-phospho-D-gl...  EC 1.1.1.49  \n",
      "5   This enzyme belongs to the family of hydrolase...  EC 3.1.1.17  \n",
      "6   It forms ribulose 5-phosphate from 6-phosphogl...  EC 1.1.1.44  \n",
      "7   transaldolase:catalyzes the following reaction...   EC 2.2.1.2  \n",
      "8   This enzyme catalyzes the oxidation of succina...   EC 1.3.5.1  \n",
      "9   It catalyzes the reversible hydration/dehydrat...   EC 4.2.1.2  \n",
      "10  It reversibly catalyzes the oxidation of malat...  EC 1.1.1.37  \n",
      "11  It catalyzes the condensation reaction of the ...   EC 2.3.3.1  \n",
      "   id               name                                        description\n",
      "0   1         glycolysis  metabolic pathway that converts glucose C6H12O...\n",
      "1   2  pentose phosphate  metabolic pathway parallel to glycolysis, gene...\n",
      "2   3          TCA cycle  a series of chemical reactions used by all aer...\n",
      "    eid  pid  priority\n",
      "0     1    1         1\n",
      "1     2    1         2\n",
      "2     3    1         3\n",
      "3     4    1         4\n",
      "4     4    2         0\n",
      "5     5    2         1\n",
      "6     5    1         0\n",
      "7     6    2         2\n",
      "8     7    2         3\n",
      "9     8    2         4\n",
      "10    9    3         1\n",
      "11   10    3         2\n",
      "12   11    3         3\n",
      "13   11    1         0\n",
      "14   12    3         4\n"
     ]
    }
   ],
   "source": [
    "# print table\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "print (pd.read_sql_query(\"SELECT * FROM gene;\",conn))\n",
    "print (pd.read_sql_query(\"SELECT * FROM enzyme;\", conn))\n",
    "print (pd.read_sql_query(\"SELECT * FROM pathway;\", conn))\n",
    "print (pd.read_sql_query(\"SELECT * FROM en_pa;\", conn))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   id                               name\n",
      "0   5  Glucose-6-phosphate dehydrogenase\n",
      "1   6                   Gluconolactonase\n",
      "2   7   6-phosphogluconate dehydrogenase\n",
      "3   8                      Transaldolase\n"
     ]
    }
   ],
   "source": [
    "# view by enzyme priority in each pathway SORTED BY ORDER!!\n",
    "\n",
    "print(pd.read_sql_query(\"SELECT enzyme.id, enzyme.name FROM enzyme INNER JOIN en_pa on enzyme.id = en_pa.eid WHERE en_pa.pid = 2 and en_pa.priority>0 ORDER BY en_pa.priority ASC ;\", conn))\n",
    "# sort enzymes by iwhere pid=1 and have priority information(priority>0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
